File: coding-test/java-kotlin/src/kr/co/programmers/java/summer_winter_coding_2018/Budget.java
package kr.co.programmers.java.summer_winter_coding_2018;

import java.util.Arrays;

public class Budget {

    public static void main(String[] args) {
        int[] d = {1, 3, 2, 5, 4};
        int budget = 9;
        System.out.println(new Solution().solution(d, budget));
    }

    static class Solution {
        public int solution(int[] d, int budget) {
            int answer = 0;

            Arrays.sort(d);

            for(int m : d){
                if(m > budget){
                    break;
                }

                budget = budget - m;
                answer++;
            }

            return answer;
        }
    }
}

File: practice-coding-test/javascript/src/summer-winter-coding-2018/budget.js
// @ts-check

/**
 * @param {number[]} d
 * @param {number} budget
 * @returns
 */
function solution(d, budget) {
  let answer = 0
  d = d.sort((a, b) => a - b)
  for (let i = 0; i < d.length; i += 1) {
    if (budget >= d[i]) {
      answer += 1
      budget -= d[i]
    }
  }

  return answer
}

const d = [1, 3, 2, 5, 4]
const budget = 9
console.log(solution(d, budget))
